viewing_party/main.py

```diff
@@ -0,0 +1,221 @@
+
+def create_movie(title, genre, rating): 
+    movie = dict();
+    if title:
+        movie['title'] = title;
+    else:
+        return None
+    if genre:
+        movie['genre'] = genre;
+    else:
+        return None
+    if rating:
+        movie['rating'] = rating;
+    else:
+        return None
+    return movie
+
+
+def add_to_watched(user_data, movie):   
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    watched_movie_temp_list = []
+
+    for key, values in user_data.items():
+        for movie_info in values:
+            if title == movie_info["title"]:
+                user_data["watchlist"].remove(movie_info)
+                watched_movie_temp_list.append(movie_info)
+    for watched_movie in watched_movie_temp_list:
+        user_data["watched"].append(watched_movie)
+
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+
+    average_ratings = []
+    average = []
+
+    if not user_data["watched"]:
+        return 0.0
+
+    for keys, values in user_data.items():
+        for watched_movies in values:   
+            if watched_movies["rating"] == '':
+                watched_movies["rating"] = 0.0
+            else:
+                average_ratings.append(float(watched_movies["rating"]))
+                average = sum(average_ratings)/len(average_ratings)
+        return average
+
+def get_most_watched_genre(user_data):
+
+    genre_count = []
```

Comment:
Style: It's convention to give variables that hold lists plural names:

```suggestion
    genre_counts = []
```